tutorials/xr/xr_action_map.rst

```diff
@@ -0,0 +1,260 @@
+.. _doc_xr_action_map:
+
+The XR action map
+=================
+
+Godot has an action map feature as part of the XR system. At this point in time this system is part of the OpenXR module but there are plans to encompass WebXR into this in the near future hence we call it the XR action map system in this document. It implements the built-in action map system of OpenXR mostly exactly as it is offered.
+
+The XR action map system exposes input, positional data and output for XR controllers to your game/application. It does this by exposing named actions that can be tailored to your game/application and binding these to the actual inputs and outputs on your XR devices.
+
+As the XR action map is currently part of the OpenXR module, OpenXR needs to be enabled in your project settings to expose it:
+
+.. image:: img/openxr_settings.png
+
+You will then find the XR Action Map interface in the bottom of the screen:
+
+.. image:: img/xr_action_map.webp
+
+.. note::
+  Godot's built-in input system has many things in common with the XR action map system. In fact our original idea was to add functionality to the existing input system and expose the data to the OpenXR action map system. We may revisit that idea at some point but as it turns out there were just too many problems to overcome. To name a few:
+
+    * Godot's input system mainly centers around button inputs, XR adds triggers, axis, poses and haptics (output) into the mix. This would greatly complicate the input system with features that won't work for normal controllers or contrast with the current approach. It was felt this would lead to confusion for the majority of Godot users.
+    * Godot's input system works with raw input data that is parsed and triggers emitting actions, this input data is made available to the end user. OpenXR completely hides raw data and does all the parsing for us, we only get access to already parsed action data. This inconsistency is likely to lead to bugs when an unsuspecting user tries to use an XR device as a normal input device.
+    * Godot's input system allows changes to what inputs are bound to actions in runtime, OpenXR does not.
+    * Godot's input system is based on device ids which are meaningless in OpenXR.
+
+  This does mean that a game/application that mixes traditional inputs with XR controllers will have a separation. For most applications either one or the other is used and this is not seen as a problem. In the end, it's a limitation of the system.
```

Comment:
What exactly is meant here by "a separation" does that mean a project at runtime can only use one or the other?

Reply:
It means that you'll be implementing two separate systems. You'll be handling keyboard/mouse/HOTAS input through Godots existing input system, and XR controllers through the action system. You can't create an interface where the user can optionally map an action to either traditional or XR inputs.

In practice this generally isn't a problem.

The majority of XR games only feature XR input as most now follow the traditional "Quest" like model of a headset with two XR controllers in hand.

Those that don't, like driving sims or flight sims, go the other way ignoring XR controllers in favor of racing wheels and HOTAS.

There are a very tiny number of use cases were people want to mix and match but I do want to underline here that the choice was made for us. OpenXR takes over input binding and parsing and has closed the door to easily supporting this scenario and that as I understand it, was a conscious decision.